contracts/mesh-provider/src/contract.rs

```diff
@@ -253,6 +271,56 @@ pub fn execute_unbond(
     Ok(Response::new().add_message(msg))
 }
 
+// HACK this implementation of claiming rewards is not performant or robust
+// It is intended for proof of concept only.
+pub fn execute_claim_rewards(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    validator: String,
+) -> Result<Response, ContractError> {
+    let delegator = deps.api.addr_validate(info.sender.as_str())?;
```

Comment:
info.sender comes from Go module, is already validated. No need to do this.

`let delgator = info.sender;` is sufficient